app/controllers/sessions/session.rb

```diff
@@ -0,0 +1,28 @@
+# Same as /sessions/new
+get '/login' do
+  erb :'/users/login'
+end
+
+# Same as post /sessions
+post '/login' do
+  user = User.find_by(username: params[:user][:username])
+  password = params[:user][:password]
+
+  if user && User.authenticate(params[:user][:username], password)
+    session[:user_id] = user.id
+    redirect "/"
+  else
+    redirect '/login'
+  end
+end
+
+# Same as delete /sessions/:id
+get '/logout' do
```

Comment:
/logout is an alias to /sessions/destroy which should be the following route:
delete '/logout'

There should be a form with a button "delete" that submits to /logout. Note there should be a hidden attribute saying it's a DELETE operation.